packages/handlersjs-http/lib/services/sync.service.spec.ts

```diff
@@ -0,0 +1,288 @@
+    httpStatus: number;
+    storages?: number[];
+  }
+
+  // mocks the fetch() function with http responses according to the peerResponseMock interface
+  const mockWithStorages = (... responseMocks: peerResponseMock[]) => {
+
+    fetchMock.mockImplementation(async (url, options) => {
+
+      latestModifiedSince = options?.headers['If-Modified-Since'];
+
+      for (const mockInfo of responseMocks) {
+
+        if (mockInfo.peer === url) {
+
+          return new Response(JSON.stringify(mockInfo.storages ? mockInfo.storages : []),
+            { status: mockInfo.httpStatus, headers });
+
+        }
+
+      }
+
+      return new Response('[]', { status: 200, headers });
+
+    });
+
+  };
+
+  describe('sync()', () => {
+
+    it('resolves when there are no peers', async () => {
+
+      await store.set('peers', new Set());
+      await expect(syncService.sync()).resolves.toBe(syncService);
+
+    });
+
+    it('resolves when peers are undefined', async () => {
+
+      await store.delete('peers');
+      await expect(syncService.sync()).resolves.toBe(syncService);
+
+    });
+
+    it('sends a request to all peers in the store', async () => {
+
+      await syncService.sync();
+      fetchMock.mock.calls.forEach((call) => expect(peers.has(call[0].toString())).toBe(true));
+
+    });
+
+    describe('If-Modified-Since behavior', () => {
+
+      it('doesn\'t have the If-Modified-Since header on the first request', async () => {
+
+        await syncService.sync();
+        expect(latestModifiedSince).toBeUndefined();
+
+      });
+
+      it('has the If-Modified-Since header from the second request and onwards', async () => {
+
+        await syncService.sync();
+        expect(latestModifiedSince).toBeUndefined();
+        await syncService.sync();
+        expect(latestModifiedSince).toBeDefined();
+        await syncService.sync();
+        expect(latestModifiedSince).toBeDefined();
+
+      });
+
+      it('has the correct modifiedSince header', async () => {
+
+        let beforeFirstSync = Date.now();
+        beforeFirstSync -= beforeFirstSync % 1000; // If-Modified-Since header value rounds down to the second
+        await syncService.sync();
+        const afterFirstSync = Date.now();
+
+        await syncService.sync();
+        const firstSync = new Date(latestModifiedSince).getTime();
+
+        expect(beforeFirstSync <= firstSync && firstSync <= afterFirstSync).toBe(true);
+
+      });
+
+      it('updates the If-Modified-Since header correctly', async () => {
+
+        await syncService.sync();
+        await syncService.sync();
+        const firstSync = new Date(latestModifiedSince).getTime();
+
+        const nextSync = firstSync + 5200;
+        const nextDate = new Date(nextSync);
+
+        const mockDate = jest
+          .spyOn(global, 'Date')
+          .mockImplementation(() => nextDate as unknown as string);
+        // https://stackoverflow.com/questions/28504545/how-to-mock-a-constructor-like-new-date
```

Comment:
Sure you need to test this by mocking Date?

Reply:
I'm not sure what would be a better option

Reply:
Hm, okay, I don't have an immediate better idea. But you could probably call `.toString()` on that `Date` instead of casting it, no?

Reply:
solved by using jest-date-mock